src/core/mac/mac.cpp

```diff
@@ -1154,6 +1156,11 @@ ThreadError Mac::ProcessReceiveSecurity(Frame &aFrame, const Address &aSrcAddr,
             mKeyManager.SetCurrentKeySequence(keySequence);
         }
     }
+    else if (keyIdMode == Frame::kKeyIdMode2)
+    {
+        // reset frame counter
+        mKeyIdMode2FrameCounter = 0;
```

Comment:
Strictly speaking, the device needs to maintain two separate frame counters, one for transmitted frames and another for received frames.  The intent behind Section 7.2.1.3 is to disable the frame counter check on received frames.  Resetting the receive frame counter to zero on each received frame logically does that.  Of course, this implementation can simply ignore the frame counter field.

Additionally, the Frame Counter on transmitted frames should not matter.  Is there a reason this PR changes that to an incrementing value?